pkg/automationconfig/automation_config_secret.go

```diff
@@ -11,6 +11,22 @@ import (
 
 const ConfigKey = "cluster-config.json"
 
+// ReadFromSecret returns the AutomationConfig present in the given Secret. If the Secret is not
+// found, it is not considered an error and an empty AutomationConfig is returned.
+func ReadFromSecret(secretGetter secret.Getter, secretNsName types.NamespacedName) (AutomationConfig, error) {
+	acSecret, err := secretGetter.GetSecret(secretNsName)
+	if err != nil && !apiErrors.IsNotFound(err) {
+		return AutomationConfig{}, err
+	}
+
+	// the secret not existing is perfectly fine, this will happen during the first reconciliation.
+	if apiErrors.IsNotFound(err) {
+		return AutomationConfig{}, nil
+	}
```

Comment:
[q] what do you think of writing this as:

```go
	if err != nil {
		// the secret not existing is perfectly fine, this will happen during the first reconciliation.
     	if apiErrors.IsNotFound(err) {
			return AutomationConfig{}, nil
		} else  {
			return AutomationConfig{}, err
		}
}
```

Or maybe using [IgnoreNotFound](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/interfaces.go#L131) (although this would require changing `secret.Getter` to a client, or cast it - or write that method in our own error package)

Reply:
ah I think we can use `IgnoreNotFound` to make it quite concise! I generally prefer to avoid using the `else` keyword if possible to make the main logic as flat as possible. We could do something like 

```golang

func ReadFromSecret(secretGetter secret.Getter, secretNsName types.NamespacedName) (AutomationConfig, error) {
	acSecret, err := secretGetter.GetSecret(secretNsName)
	if err != nil {
		return AutomationConfig{}, client.IgnoreNotFound(err)
	}
	return FromBytes(acSecret.Data[ConfigKey])
}
```